x-pack/plugins/apm/server/lib/errors/get_error_groups.ts

```diff
@@ -147,15 +147,15 @@ export async function getErrorGroups({
     const source = bucket.sample.hits.hits[0]._source;
     const message =
       idx(source, _ => _.error.log.message) ||
-      idx(source, _ => _.error.exception.message);
+      idx(source, _ => _.error.exception[0].message);
```

Comment:
I'm probably missing something but how does this work for old data where `exception` is still a single object and not an array?

Reply:
AFAICT, all errors from existing data currently show "No stacktrace available", but the linked ticket says:
>the minimal change would be update the UI to expect error.exception to be either an object or an array of objects, and in the latter cause just take the first object.

Reply:
True that issue is not very clear. I've added @roncohen's comment now:

> this is something we'd do now in 7.0 to make sure the UI doesn't have to cater for multiple types in a way that allows us to do the rest later without breaking backwards compatibility later?
we're going to ask people to reindex everything for 7.0, so we could also change the existing data from a single object to an array.
Seems like a fairly minor thing both reindex wise and UI wise, so i'd be happy to try to see if we can get it done for 7.0

https://github.com/elastic/apm/issues/40#issuecomment-455497016

Reply:
(I didn't initially add the link to the original issue due to out policy about not linking to private repos, but too much context is lost by just pasting in comments, so now I'm linking to it.

Reply:
tldr: afaict we only have to support exceptions as arrays (starting from 7.0). Please confirm @roncohen

Reply:
that is correct. All data will be reindexed and `error.exception` will have always have an array (see  https://github.com/elastic/apm-server/issues/1807 third item)

Reply:
Sounds good. Is there a UI testing plan for the Great Reindexing?

Reply:
@graphaelli has been working on a framework to help test the 7.0 changes. I don't know if that'll be applicable for UI testing. Thoughts @graphaelli ?